src/app/modules/item/services/item-datasource.service.ts

```diff
@@ -22,11 +22,15 @@ export interface ItemData { route: FullItemRoute, item: Item, breadcrumbs: Bread
 @Injectable()
 export class ItemDataSource implements OnDestroy {
 
+  private readonly fetchOperation$ = new ReplaySubject<FullItemRoute>(1); // trigger item fetching
+  private readonly refresh$ = new Subject<void>();
+
   /* state to put outputted */
-  private state = new BehaviorSubject<FetchState<ItemData>>(fetchingState());
-  state$ = this.state.asObservable();
+  readonly state$ = this.fetchOperation$.pipe(
+    switchMap(item => this.fetchItemData(item)),
+    mapToFetchState({ resetter: merge(this.refresh$, this.fetchOperation$) })
+  );
```

Comment:
When refactoring the group data source, I had to add a `shareReplay(1)` operator
As is, if a component subscribes to the datasource _after_ the fetch has succeeded, the component won't have any value

I tested switching between edit and normal mode on [this item](https://dev.algorea.org/branch/use-fetchstate-in-itemdatasource/en/#/activities/by-id/1980584647557587953;path=4702,4102;attempId=0/details), it apparently works but I suspect it is because the item is refetched.

Should this `shareReplay(1)` + double call issue be addressed in a another issue+PR ?

Reply:
Ok, I can do this.
What you call "double call" is the fact that get-item is called in item-edit (why probably it shouldn't, except to be sure we work on the same version)?
Actually, I am thinking about moving this item-datasource to something private to item-by-id and then just pass the item by input. This is a remaining from a time where I had not better way to pass the item as the subpage are served by routing.

Reply:
fixed, and it fix multiple calls

Reply:
Weird, I have 2 cancelled calls at item init, do you have an idea of where it could come from ?
![image](https://user-images.githubusercontent.com/16257159/131673221-a150f369-b868-4a60-b77e-a352cd3d7f1b.png)